snuba/query/conditions.py

```diff
@@ -48,6 +48,26 @@ class BooleanFunctions:
     OR = "or"
 
 
+def __set_condition(
+    alias: Optional[str], function: str, lhs: Expression, rhs: Sequence[Literal]
+) -> Expression:
+    return binary_condition(
+        alias, function, lhs, FunctionCall(None, "tuple", tuple(rhs)),
+    )
```

Comment:
Would it be worth adding a `Tuple` or `Set` subtype of `Expression` to handle this case? I feel like this is going to be pretty common. ("Maybe later" is a reasonable answer here.)

Reply:
I think it would make sense but as a subclass of FunctionCall since this is a function call and formats as a function call in Clickhouse (same reasoning we do for conditions). So that, when processing the query, this expression can be identified either as a tuple, or as a function call depending what the processing code wants to do. 
Does that make sense ?

Reply:
Yeah, I think that makes a lot of sense and seems like the best of both worlds.

Reply:
It actually did not work. When I extended the dataclass, I could not mark the new one as dataclass because I needed a different constructor, and that failed when running dataclass.replace. 
For now I added a helper method in a DSL module. Will get back to this to see if I can make it a class in the AST.

The only downside I can think about though is that the fact that `tuple` is a function is a bit of a Clickhouse specific detail. (But is the idea that all conditions are functions).